### PR TITLE
Github actions - versiongit information from tags

### PR DESCRIPTION
**Pull request by mpatrou**
## Description of work:
split checkout and micromamba and include tag information only for deployed packages

Check all that apply:
- [ ] added [release notes](https://github.com/neutrons/drtsans/blob/next/docs/release_notes.rst) (if not, provide an explanation in the work description)
- [ ] updated documentation
- [ ] Source added/refactored
- [ ] Added unit tests
- [ ] Added integration tests
- [ ] Verified that tests requiring the /SNS and /HFIR filesystems pass without fail

**References:**
- Links to IBM EWM items: [CI versioningit reports incorrect software version](https://ornlrse.clm.ibmcloud.com/ccm/resource/itemName/com.ibm.team.workitem.WorkItem/9184)
- Links to related issues or pull requests:

## Check list for the reviewer
- [ ] [release notes](https://github.com/neutrons/drtsans/blob/next/docs/release_notes.rst) updated, or an explanation is provided as to why release notes are unnecessary
- [ ] best software practices
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date
- [ ] code comments added when explaining intent

### Execution of tests requiring the /SNS and /HFIR filesystems
It is strongly encouraged that the reviewer runs the following tests in their local machine
because these tests are not run by the GitLab CI. It is assumed that the reviewer has the /SNS and /HFIR filesystems
remotely mounted in their machine.

```bash
cd /path/to/my/local/drtsans/repo/
git fetch origin merge-requests/<MERGE_REQUEST_NUMBER>/head:mr<MERGE_REQUEST_NUMBER>
git switch mr<MERGE_REQUEST_NUMBER>
conda activate <my_drtsans_dev_environment>
pytest -m mount_eqsans ./tests/unit/ ./tests/integration/
```
In the above code snippet, substitute `<MERGE_REQUEST_NUMBER>` for the actual merge request number. Also substitute
`<my_drtsans_dev_environment>` with the name of the conda environment you use for development. It is critical that
you have installed the repo in this conda environment in editable mode with `pip install -e .` or `conda develop .`

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -50,7 +50,34 @@ jobs:
       - name: Build the docs
         run: |
           cd docs && make html
-      - name: Build conda package
+
+  package:
```
Comment (jmborr) [inline .github/workflows/test.yml:54]: This should not be a different job than `tests` because only the last step `Build Conda package` is required. The python wheels is not required and the other steps duplicate those of job `tests`.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -50,7 +50,34 @@ jobs:
...
+          miniconda-version: latest
+          mamba-version: "*"
+          environment-file: environment.yml
+          lfs: true
+          submodules: recursive
+      - name: Build Wheel
+        run: |
+          python -m build --wheel --no-isolation
+          check-wheel-contents dist/drtsans*.whl
+      - name: Build Conda package
```
Comment (jmborr) [inline .github/workflows/test.yml:80]: step `Build Conda package` should run only for pull requests or branch `next`, but not for `qa`,`main`, or any tags because these already run in workflow `package.yml`

**Review by jmborr — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (mpatrou) [reply .github/workflows/test.yml:54]: It cannot be in the same tests job, because of git lfs. Since the git lfs data repo is pulled, when conda build is triggered, versiongit reports the version from the git-lfs data repo over this one.

File: .github/workflows/test.yml (hunk 2 above)
Comment (mpatrou) [reply .github/workflows/test.yml:80]: This just ensures the code can be wrapped as a python and conda package after every PR. We have it in other projects, to make sure that changes like including a new file format, will be included properly, before we make any package. However, if it something you don't need here, we can discard the whole PR.

Comment (jmborr) [reply .github/workflows/test.yml:80]: It is useful to know that the conda package builds from any pull request and from branch `next`. Additional functionality is not necessary or already taken care in workflow `package.yml`

File: docs/release_notes.rst [hunk 3]
```diff
@@ -31,6 +31,7 @@ Release Notes
   - PR #1022: Generator of simulated runs for testing
   - PR #1021: stub for the polarization module
   - PR #1019: Fixes an edge case bug in creating the sample logs periodic index log where the number of entries is less than the number of times
+  - PR #1023: Github actions - conda build in tests workflow and versiongit information from tags
```
Comment (jmborr) [inline docs/release_notes.rst]: it's a stack. Newest go to at  top

File: docs/release_notes.rst [hunk 4]
```diff
@@ -31,6 +31,7 @@ Release Notes
   - PR #1022: Generator of simulated runs for testing
   - PR #1021: stub for the polarization module
   - PR #1019: Fixes an edge case bug in creating the sample logs periodic index log where the number of entries is less than the number of times
+  - PR #1023: Github actions - conda build versiongit information from tags
```
Comment (jmborr) [inline docs/release_notes.rst:34]: it's a stack. Latest entry goes at the top

**Review by jmborr — CHANGES_REQUESTED** [review]

**Review by jmborr — APPROVED** [review]
